### PR TITLE
Add assigned permset preflight

### PR DESCRIPTION
**Pull request by jstvz**
# Critical Changes

# Changes
- Added a new preflight check `get_assigned_permission_sets`, which provides a list of API Names of permission sets assigned to the running user.
# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/preflight/permsets.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class GetPermissionSetAssignments(BaseSalesforceApiTask):
+    def _run_task(self):
+        query = f"SELECT PermissionSet.Name FROM PermissionSetAssignment WHERE AssigneeId = '{self.org_config.user_id}'"
+        self.return_values = [
+            result["PermissionSet"]["Name"]
+            for result in self.sf.query(query)["records"]
```
Comment (davidmreed) [inline cumulusci/tasks/preflight/permsets.py:9]:
```suggestion
            for result in self.sf.query_all(query)["records"]
```

Got burned on this one just the other day.

File: cumulusci/tasks/preflight/tests/test_permset_preflights.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    "PermissionSet": {
+                        "Label": "Einstein Analytics Plus Admin",
+                        "Name": "EinsteinAnalyticsPlusAdmin",
+                    },
+                },
+            ],
+        }
+        task()
+
+        task._init_api.return_value.query.assert_called_once_with(
```
Comment (davidmreed) [inline cumulusci/tasks/preflight/tests/test_permset_preflights.py:32]:
```suggestion
        task._init_api.return_value.query_all.assert_called_once_with(
```

**Review by davidmreed — CHANGES_REQUESTED** [review]

File: cumulusci/tasks/preflight/permsets.py (hunk 1 above)
Comment (jstvz) [reply cumulusci/tasks/preflight/permsets.py:9]: Thanks, I keep forgetting this lesson.

File: cumulusci/cumulusci.yml [hunk 3]
```diff
@@ -252,6 +252,10 @@ tasks:
         description: Retrieves a list of the currently available Permission Set License definition keys
         class_path: cumulusci.tasks.preflight.licenses.GetAvailablePermissionSetLicenses
         group: Salesforce Preflight Checks
+    get_assigned_permission_sets:
+        description: Retrieves a list of the label and names of any permission sets assigned to the running user.
```
Comment (davidmreed) [inline cumulusci/cumulusci.yml:256]: Sorry, one more proposed tweak:

```suggestion
        description: Retrieves a list of the names of any permission sets assigned to the running user.
```

**Review by davidmreed — CHANGES_REQUESTED** [review]

**Review by davidmreed — APPROVED** [review]

**Pull request merged**